hermes-schema/src/main/java/pl/allegro/tech/hermes/schema/SchemaRepository.java

```diff
@@ -39,7 +39,18 @@ public SchemaRepository(SchemaVersionsRepository schemaVersionsRepository,
      * from underlying schema repository.
      */
     public CompiledSchema<Schema> getKnownAvroSchemaVersion(Topic topic, SchemaVersion version) {
-        return compiledAvroSchemaRepository.getSchema(topic, version);
+        return getCompiledSchemaAtVersion(topic, version);
+    }
+
+    private CompiledSchema<Schema> getCompiledSchemaAtVersion(Topic topic, SchemaVersion latestVersion) {
+        try {
+            return compiledAvroSchemaRepository.getSchema(topic, latestVersion);
+        } catch (Exception e) {
+            if (e instanceof CouldNotLoadSchemaException) {
```

Comment:
\[PMD\] INFO: An instanceof check is being performed on the caught exception\.  Create a separate catch clause for this exception type\.